gadget-container/entrypoint.go

```diff
@@ -0,0 +1,290 @@
+	}
+
+	err = copyFile(destinationPath, "/opt/hooks/nri/nrigadget", 0640)
+	if err != nil {
+		return fmt.Errorf("copying: %w", err)
+	}
+
+	configPath := "/opt/hooks/nri/conf.json"
+
+	content, err := os.ReadFile(configPath)
+	if err == nil {
+		var object map[string]interface{}
+
+		err := json.Unmarshal(content, &object)
+		if err != nil {
+			return fmt.Errorf("unmarshalling JSON %s: %w", configPath, err)
+		}
+
+		add := map[string]string{ "type": "nrigadget" }
+
+		_, ok := object["plugins"]
+		if !ok {
+			object["plugins"] = []interface{}{ add }
+		} else {
+			plugins, ok := object["plugins"].([]interface{})
+			if !ok {
+				return fmt.Errorf("JSON object is not correct: %+v", object)
+			}
+
+			object["plugins"] = append(plugins, add)
+		}
+
+		content, err = json.Marshal(object)
+		if err != nil {
+			return fmt.Errorf("marshalling JSON: %w", err)
+		}
+
+		err = os.WriteFile(configPath, content, 0640)
+		if err != nil {
+			return fmt.Errorf("writing %s: %w", configPath, content)
+		}
+	} else {
+		destinationPath := filepath.Join(host.HostRoot, "etc/nri")
+		os.MkdirAll(destinationPath, 0755)
+		if err != nil {
+			return fmt.Errorf("creating %s: %w", destinationPath, err)
+		}
+
+		err := copyFile(destinationPath, configPath, 0640)
+		if err != nil {
+			return fmt.Errorf("copying: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	if _, err := os.Stat(filepath.Join(host.HostRoot, "/bin")); os.IsNotExist(err) {
+		log.Fatalf("%s must be executed in a pod with access to the host via /host", os.Args[0])
+	}
+
+	prettyName, err := getPrettyName()
+	if err != nil {
+		log.Warnf("os-release information not available. Some features could not work")
+	}
+
+	log.Infof("OS detected: %s", prettyName)
+
+	kernelRelease, err := getKernelRelease()
+	if err != nil {
+		log.Fatalf("getting kernel release: %w", err)
+	}
+
+	log.Infof("Kernel detected: %s", kernelRelease)
+
+	log.Infof("Gadget Image: %s", os.Getenv("GADGET_IMAGE"))
+
+	log.Info("Deployment options:")
+	for _, variable := range os.Environ() {
+		if strings.HasPrefix(variable, "INSPEKTOR_GADGET_OPTION_") {
+			log.Infof("%s", variable)
+		}
+	}
+
+	log.Infof("Inspektor Gadget version: %s", os.Getenv("INSPEKTOR_GADGET_VERSION"))
+
+	path := "/proc/self/cgroup"
+	content, err := os.ReadFile(path)
+	if err != nil {
+		log.Fatalf("reading %s: %w", path, err)
+	}
+
+	crio := false
+	if regex.Match(content){
+		log.Infof("CRI-O detected.")
+		crio = true
+	}
+
+	hookMode := os.Getenv("INSPEKTOR_GADGET_OPTION_HOOK_MODE")
+	if (hookMode == "auto" || hookMode == "") && crio {
+		log.Info("Hook mode CRI-O detected")
+		hookMode = "crio"
+	}
+
+	switch hookMode {
+	case "crio":
+		err := installCRIOHooks()
+		if err != nil {
+			log.Fatalf("installing CRIO hooks: %w", err)
+		}
+	case "nri":
+		err := installNRIHooks()
+		if err != nil {
+			log.Fatalf("installing NRI hooks: %w", err)
+		}
+	}
+
+	gadgetTracerManagerHookMode := "auto"
+	switch hookMode {
+	case "crio", "nri":
```

Comment:
nit: I wonder if it makes sense to use `const` for these since we already use them in multiple places [1] [2], not sure about the best place! 

1. https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgettracermanager/gadgettracermanager.go#L282
2. https://github.com/inspektor-gadget/inspektor-gadget/blob/main/cmd/kubectl-gadget/deploy.go#L89